web3_proxy/src/referral_code.rs

```diff
@@ -0,0 +1,25 @@
+use anyhow::{self, Result};
+use thread_fast_rng::rand::distributions::Alphanumeric;
+use thread_fast_rng::rand::{Rng, thread_rng};
+
+pub struct ReferralCode(pub String);
+
+impl Default for ReferralCode {
+    fn default() -> Self {
+        let mut rng = thread_rng();
+        let chars: String = (0..32).map(|_| rng.sample(Alphanumeric) as char).collect();
```

Comment:
it's not exactly readable, but im assuming here that codes are shared via link (which they usually are to my knowledge..) can look into more readable formats if you prefer

Reply:
Use ULIDs. The crate is already used by web3-proxy. They are short and readable.

Reply:
https://github.com/llamanodes/web3-proxy/pull/35/commits/edd9371284554de7e5ed083cd0d03169a76df791